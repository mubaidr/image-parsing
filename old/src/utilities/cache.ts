import Cache from '../utilities/@classes/Cache'

const cache = new Cache()

export { cache }
