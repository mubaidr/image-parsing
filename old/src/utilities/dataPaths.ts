import os from 'os'
import path from 'path'

const isDev: boolean = process.env.NODE_ENV === 'development'
const tmp: string = isDev ? path.resolve('.tmp') : os.tmpdir()
const home: string = path.join(os.homedir(), 'desktop')

const dataPaths = {
  design: path.resolve('__tests__', '_data', 'design.qr.svg'),
  designBarcode: path.resolve('__tests__', '_data', 'design.svg'),
  home,
  images: path.resolve('__tests__', '_data', 'images-qrcode'),
  imagesBarcode: path.resolve('__tests__', '_data', 'images-barcode'),
  key: path.resolve('__tests__', '_data', 'key.xlsx'),
  keyImage: path.resolve('__tests__', '_data', 'key.jpg'),
  questionsModel: isDev
    ? path.resolve('src/data/questions-model.json')
    : path.resolve('dist/data/questions-model.json'),
  result: path.resolve('__tests__', '_data', 'result.xlsx'),
  resultCompiled: path.resolve('__tests__', '_data', 'resultCompiled.xlsx'),
  root: path.resolve('.'),
  temp: tmp,
  tmp,
}

export { dataPaths }
