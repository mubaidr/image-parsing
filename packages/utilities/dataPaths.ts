import os from 'os';
import path from 'path';

const tmp = os.tmpdir();

export const DataPaths = {
  design: path.resolve('tests', '_data', 'design.qr.svg'),
  designBarcode: path.resolve('tests', '_data', 'design.svg'),
  home: os.homedir(),
  images: path.resolve('tests', '_data', 'images-qrcode'),
  imagesBarcode: path.resolve('tests', '_data', 'images-barcode'),
  key: path.resolve('tests', '_data', 'key.xlsx'),
  keyImage: path.resolve('tests', '_data', 'key.jpg'),
  result: path.resolve('tests', '_data', 'result.xlsx'),
  resultCompiled: path.resolve('tests', '_data', 'resultCompiled.xlsx'),
  root: path.resolve('.'),
  temp: tmp,
  tmp,
};
