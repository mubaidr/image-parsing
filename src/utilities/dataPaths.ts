import os from 'os'
import path from 'path'

const isDev: boolean = process.env.NODE_ENV === 'development'

const tmp: string = isDev ? path.resolve('.tmp') : os.tmpdir()
const home: string = path.join(os.homedir(), 'desktop')

const dataPaths = {
  design: path.resolve('_test_data', 'design.qr.svg'),
  designBarcode: path.resolve('_test_data', 'design.svg'),
  home,
  images: path.resolve('_test_data', 'images-qrcode'),
  imagesBarcode: path.resolve('_test_data', 'images-barcode'),
  key: path.resolve('_test_data', 'key.xlsx'),
  keyImage: path.resolve('_test_data', 'key.jpg'),
  questionsModel: path.resolve('src/data/questions-model.json'),
  result: path.resolve('_test_data', 'result.xlsx'),
  root: path.resolve('.'),
  temp: tmp,
  tmp,
}

export { dataPaths }
