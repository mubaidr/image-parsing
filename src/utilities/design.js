const cheerio = require('cheerio')
const fs = require('fs')

const OPTIONS = require('./question-options')

/**
 * Extracts position & dimensions of objects from SVG design File
 *
 * @param {String} designFilePath Path to the svg design file
 * @returns {Object} JSON object
 */
async function getDesignData(dir) {
  const designData = {
    questions: {},
  }
  const ROLL_NO_PATTERN = new RegExp(/rollnobarcode/gi)
  const QUESTION_PATTERN = new RegExp(/(q[1-9][0-9]?[ad])\b/gi)

  const $ = cheerio.load(fs.readFileSync(dir, 'utf8'))
  const svgViewBox = $('svg')[0].attribs.viewBox.split(' ')

  designData.width = Math.ceil(svgViewBox[2] - svgViewBox[0])
  designData.height = Math.ceil(svgViewBox[3] - svgViewBox[1])

  let gotRollNo = false
  let x
  let y
  let rx
  let ry
  let width
  let height
  let left
  let top

  const groups = $('g')
  for (let i = 0; i < groups.length; i += 1) {
    const group = groups[i]

    const title =
      $(group)
        .find('title')
        .first()
        .html()
        .trim()
        .toUpperCase() || ''

    if (!title) continue

    const isQuestionGroup = QUESTION_PATTERN.test(title)
    const isRollNoGroup =
      isQuestionGroup || gotRollNo ? false : ROLL_NO_PATTERN.test(title)

    if (!(isQuestionGroup || isRollNoGroup)) continue

    const transform = $(group)
      .attr('transform')
      .replace(/(translate)|\(|\)/gi, '')
      .split(',')
      .map(val => parseInt(val, 10) || 0)

    const rect = $(group)
      .find('rect')
      .first()

    left = parseInt(rect.attr('x'), 10) || 0
    top = parseInt(rect.attr('y'), 10) || 0

    rx = parseInt(rect.attr('rx'), 10) || 0
    ry = parseInt(rect.attr('ry'), 10) || 0

    x = left - rx + transform[0]
    y = top - ry + transform[1]

    width = parseInt(rect.attr('width'), 10) || 0 + rx
    height = parseInt(rect.attr('height'), 10) || 0 + ry

    if (isQuestionGroup) {
      const optionTitle = title.slice(-1)
      const questionNumber = title.slice(0, -1)

      if (!designData.questions[questionNumber]) {
        designData.questions[questionNumber] = {}
      }

      if (optionTitle === OPTIONS.A) {
        designData.questions[questionNumber].x1 = x - 1
        designData.questions[questionNumber].y1 = y - 1
      } else {
        // donot check last option to suport any number of options
        designData.questions[questionNumber].x2 = x + width + 1
        designData.questions[questionNumber].y2 = y + height + 1
      }
    } else if (isRollNoGroup) {
      designData.rollNo = { x1: x, y1: y, x2: x + width, y2: y + height }
      gotRollNo = true
    }
  }

  return designData
}

module.exports = {
  getDesignData,
}
