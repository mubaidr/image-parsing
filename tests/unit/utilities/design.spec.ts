import { DataPaths } from '@/utilities/dataPaths'
import { getDesignData } from '@/utilities/design'

describe('getDesignData', () => {
  test('works', async () => {
    const designData = await getDesignData(DataPaths.designBarcode)

    expect(designData).toMatchSnapshot()
  })
})
