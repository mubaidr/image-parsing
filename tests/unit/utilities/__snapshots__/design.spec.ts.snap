// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDesignData works 1`] = `
Object {
  "code": Object {
    "height": 54,
    "width": 129,
    "x": 25,
    "y": 68,
  },
  "height": 841,
  "isQrCode": false,
  "questions": Object {
    "q1": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 478,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 478,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 478,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 478,
      },
    },
    "q10": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 718,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 718,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 718,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 718,
      },
    },
    "q11": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 744,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 744,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 744,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 744,
      },
    },
    "q12": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 771,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 771,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 771,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 771,
      },
    },
    "q13": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 478,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 478,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 478,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 478,
      },
    },
    "q14": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 504,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 504,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 504,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 504,
      },
    },
    "q15": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 531,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 531,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 531,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 531,
      },
    },
    "q16": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 558,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 558,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 558,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 558,
      },
    },
    "q17": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 584,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 584,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 584,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 584,
      },
    },
    "q18": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 611,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 611,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 611,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 611,
      },
    },
    "q19": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 638,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 638,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 638,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 638,
      },
    },
    "q2": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 504,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 504,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 504,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 504,
      },
    },
    "q20": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 664,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 664,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 664,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 664,
      },
    },
    "q21": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 691,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 691,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 691,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 691,
      },
    },
    "q22": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 718,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 718,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 718,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 718,
      },
    },
    "q23": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 744,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 744,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 744,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 744,
      },
    },
    "q24": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 159,
        "y": 771,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 182,
        "y": 771,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 205,
        "y": 771,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 227,
        "y": 771,
      },
    },
    "q25": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 478,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 478,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 478,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 478,
      },
    },
    "q26": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 504,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 504,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 504,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 504,
      },
    },
    "q27": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 531,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 531,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 531,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 531,
      },
    },
    "q28": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 558,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 558,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 558,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 558,
      },
    },
    "q29": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 584,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 584,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 584,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 584,
      },
    },
    "q3": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 531,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 531,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 531,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 531,
      },
    },
    "q30": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 611,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 611,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 611,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 611,
      },
    },
    "q31": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 638,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 638,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 638,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 638,
      },
    },
    "q32": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 664,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 664,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 664,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 664,
      },
    },
    "q33": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 691,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 691,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 691,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 691,
      },
    },
    "q34": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 718,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 718,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 718,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 718,
      },
    },
    "q35": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 744,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 744,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 744,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 744,
      },
    },
    "q36": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 270,
        "y": 771,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 293,
        "y": 771,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 316,
        "y": 771,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 339,
        "y": 771,
      },
    },
    "q37": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 478,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 478,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 478,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 478,
      },
    },
    "q38": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 504,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 504,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 504,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 504,
      },
    },
    "q39": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 531,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 531,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 531,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 531,
      },
    },
    "q4": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 558,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 558,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 558,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 558,
      },
    },
    "q40": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 558,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 558,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 558,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 558,
      },
    },
    "q41": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 584,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 584,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 584,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 584,
      },
    },
    "q42": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 611,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 611,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 611,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 611,
      },
    },
    "q43": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 638,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 638,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 638,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 638,
      },
    },
    "q44": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 664,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 664,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 664,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 664,
      },
    },
    "q45": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 691,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 691,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 691,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 691,
      },
    },
    "q46": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 718,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 718,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 718,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 718,
      },
    },
    "q47": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 744,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 744,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 744,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 744,
      },
    },
    "q48": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 382,
        "y": 771,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 404,
        "y": 771,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 427,
        "y": 771,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 450,
        "y": 771,
      },
    },
    "q49": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 478,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 478,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 478,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 478,
      },
    },
    "q5": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 584,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 584,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 584,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 584,
      },
    },
    "q50": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 504,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 504,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 504,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 504,
      },
    },
    "q51": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 531,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 531,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 531,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 531,
      },
    },
    "q52": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 558,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 558,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 558,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 558,
      },
    },
    "q53": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 584,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 584,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 584,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 584,
      },
    },
    "q54": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 611,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 611,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 611,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 611,
      },
    },
    "q55": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 638,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 638,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 638,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 638,
      },
    },
    "q56": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 664,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 664,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 664,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 664,
      },
    },
    "q57": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 691,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 691,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 691,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 691,
      },
    },
    "q58": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 718,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 718,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 718,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 718,
      },
    },
    "q59": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 744,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 744,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 744,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 744,
      },
    },
    "q6": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 611,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 611,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 611,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 611,
      },
    },
    "q60": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 493,
        "y": 771,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 516,
        "y": 771,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 538,
        "y": 771,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 561,
        "y": 771,
      },
    },
    "q7": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 638,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 638,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 638,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 638,
      },
    },
    "q8": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 664,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 664,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 664,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 664,
      },
    },
    "q9": Object {
      "a": Object {
        "height": 9,
        "width": 13,
        "x": 48,
        "y": 691,
      },
      "b": Object {
        "height": 9,
        "width": 13,
        "x": 71,
        "y": 691,
      },
      "c": Object {
        "height": 9,
        "width": 13,
        "x": 93,
        "y": 691,
      },
      "d": Object {
        "height": 9,
        "width": 13,
        "x": 116,
        "y": 691,
      },
    },
  },
  "rollNo": Object {
    "height": 80,
    "width": 399,
    "x": 168,
    "y": 69,
  },
  "width": 595,
}
`;
